application/templates/certificate.yaml

```diff
@@ -18,12 +18,20 @@ metadata:
   name: {{ template "application.name" . }}-certificate
   namespace: {{ template "application.namespace" . }}
 spec:
+{{- if .Values.certificate.secretName }}
   secretName: {{ .Values.certificate.secretName }}
+{{- else }}
+  secretName: {{ template "application.name" . }}-client-tls
```

Comment:
also I would suggest to take these in helpers.tpl. we just specify "certificate.secretName" here and conditions come from helpers.tpl. will be helpful to keep templates clean